pkg/kubelet/kubelet.go

```diff
@@ -1554,6 +1555,10 @@ func (kl *Kubelet) syncLoop(updates <-chan PodUpdate, handler SyncHandler) {
 			unsyncedPod = true
 		case <-time.After(kl.resyncInterval):
 			glog.V(4).Infof("Periodic sync")
+		case <-syncStatusTicker:
+			glog.V(4).Infof("Periodic status sync")
+			kl.syncStatus()
+			continue
```

Comment:
This looks better.
However, I'm not sure if that's the best we can do. Basically assume that no updates are coming to "updates" channel. Then (before your change) we expected that SyncPods() will be called every resyncInterval. This will no longer be true with your changes because every third call will be replaced with syncStatus() (or if it will be an expensive synchronous method, even more than every third call).
So in my opinion we should remove "continue" here.

Adding @vmarmol for his opinion on it.